File: src/builder/app/src/supply/item.ts
/// <reference path="../base/conceptRef.ts" />
/// <reference path="../units/dimension.ts" />
/// <reference path="itemType.ts" />
/// <reference path="stock.ts" />

class Item {
  type: ItemType;
  name: string;
  ref: string;
  stocks: Stock[];

  public static fromRaw(raw: any): Item {
    var i : Item = new Item();

    switch(raw.type) {
      case 'Fermentables':
        i.type = ItemType.Fermentables;
        break;
      case 'Hops':
        i.type = ItemType.Hops;
        break;
      case 'Yeasts':
        i.type = ItemType.Yeasts;
        break;
      case 'Miscellaneous':
        i.type = ItemType.Miscellaneous;
        break;
      case 'Dynamics':
        i.type = ItemType.Dynamics;
        break;
    }
    i.name = raw.name;
    i.ref = raw.ref;
    i.stocks = raw.stocks.map((s: any) => { return Stock.fromRaw(this, s); });
    return i;
  }
}

File: src/core/src/main/java/com/github/malavv/brewru/inventory/Item.java
package com.github.malavv.brewru.inventory;


import java.util.ArrayList;
import java.util.List;

public class Item {
  public final ItemType type;
  public final String ref;
  public final String name;
  private List<Stock> stocks;

  public Item(final ItemType type, final String ref, final String name) {
    this.ref = ref;
    this.type = type;
    this.stocks = new ArrayList<>();
    this.name = name;
  }

  public Item addStock(final Stock stock) {
    stock.item = this;
    stocks.add(stock);
    return this;
  }

  public String getName() {
    return name;
  }
}
